Serde

0ncorhynchus opened the pull request:
It is too complicated to maintain JSON outputs implemented explicitly.
This PR, therefore, introduces a `serde` dependency for JSON outputs.

This PR also makes a change of an API changes of `Board::possible_moves_as_json()`.
It returns JSON string with the below form:

```json
[
    {
        "stones": [
             {"x": 0, "y": 0},
             {"x": 1, "y": 1},
             {"x": 1, "y": 2},
        ],
        "color": 0,
    },
    ...
]
```

ToruNiina commented:
Thanks for the suggestions!
The codes look okay, but could you explain the reason why you want to change the JSON format a bit more?

0ncorhynchus commented:
Sorry, I didn't explain it. There are two reasons.

1. I avoided implementing a new struct for JSON outputs with the old format. If I had implemented a new struct, it appeared like this:
```rust
#[derive(Serialize)]
struct PossibleMove {
    x1: i8,
    y1: i8,
    x2: i8,
    y2: i8,
    color: Color,
}
```

2. The old format is not intuitive, because it contains the list of possible roots instead of moves.
I, therefore, changed the format, which contains moves like the argument of `apply_move_if_possible`.
